src/devices/Mpr121/Mpr121.cs

```diff
@@ -0,0 +1,252 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel;
+using System.Device.I2c;
+using System.Threading;
+
+namespace Iot.Device.Mpr121
+{
+    /// <summary>
+    /// Supports MPR121 Proximity Capacitive Touch Sensor Controller.
+    /// </summary>
+    public class Mpr121 : IDisposable, INotifyPropertyChanged
+    {
+        private static readonly int CHANNELS_NUMBER = Enum.GetValues(typeof(Channels)).Length;
+
+        private I2cDevice _device;
+        private Timer _timer = null;
+
+        private Dictionary<Channels, bool> _statuses;
+
+        private int _periodRefresh;
+
+        /// <summary>
+        /// Notifies about a property has been changed.
+        /// Refresh period can be changed by setting PeriodRefresh property.
+        /// </summary>
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        /// <summary>
+        /// Gets the channels statuses. The channel status is "true" if it's pressed. "False" otherwise.
+        /// </summary>
+        public IReadOnlyDictionary<Channels, bool> ChannelStatuses
+        {
+            get
+            {
+                return _statuses.ToImmutableDictionary();
+            }
+
+            private set
+            {
+                bool isStatusChanged = false;
+                foreach (Channels channel in value.Keys)
+                {
+                    if (_statuses[channel] != value[channel])
+                    {
+                        _statuses[channel] = value[channel];
+                        isStatusChanged = true;
+                    }
+                }
+
+                if (isStatusChanged)
+                {
+                    OnPropertyChanged(nameof(ChannelStatuses));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets ot sets the period in milliseconds to refresh the channels statuses.
+        /// </summary>
+        /// <remark>
+        /// Set value 0 to stop the automatically refreshing. Setting the value greater than 0 will start/update auto-refresh.
+        /// </remark>
+        public int PeriodRefresh
+        {
+            get { return _periodRefresh; }
+
+            set
+            {
+                _periodRefresh = value;
+
+                if (_periodRefresh > 0 && _timer == null)
+                {
+                    _timer = new Timer(RefreshChannelStatuses, this, TimeSpan.FromMilliseconds(_periodRefresh), TimeSpan.FromMilliseconds(_periodRefresh));
```

Comment:
does it make sense to instead read status only when user tries to access it?

Reply:
I'm afraid we can get performance issues if we will refresh the status implicitly. Let's imagine situation when user's code looks like:
```
var channel1 = mpr121.ChannelStatuses[Channels.CH_1];
var channel2 = mpr121.ChannelStatuses[Channels.CH_2];
```

In that case the channel statuses will be read twice.

Reply:
I made some refactoring and now we don't need the ChannelStatuses property anymore. We can get the channel statuses from event args in the auto-refresh mode. Otherwise, we can call ReadChannelStatuses() method which read the status and return the value.